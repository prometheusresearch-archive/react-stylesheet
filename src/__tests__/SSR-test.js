//@flow

import React from 'react';
import ReactDomServer from 'react-dom/server';
import {style, Element} from '../';

import {Spec} from '../Strategy';

let origSpec;

const patchStrategy = (spec, targetSpec, targetApplyStrategy) => {
  Object.keys(spec).forEach(fieldName => {
    const field = spec[fieldName];

    field.applyStrategy =
      targetApplyStrategy ||
      (targetSpec && targetSpec[fieldName].applyStrategy) ||
      'static';
  });
};

describe('SSR dynamic-inline strategy', () => {
  beforeAll(() => {
    origSpec = JSON.parse(JSON.stringify(Spec));

    patchStrategy(Spec, null, 'dynamic-inline');
  });

  it('hello world button Element ', () => {
    const Button = () => (
      <Element
        background="red"
        backgroundOnHover="yellow"
        color="yellow"
        colorOnHover="red"
      />
    );

    expect(ReactDomServer.renderToString(<Button />)).toMatchSnapshot();
  });

  it('hello world button Style ', () => {
    const Button = style('div', {
      base: {
        fontSize: '16px',
        lineHeight: 20,
        color: '#ffffff',
        textDecoration: 'none',
        backgroundColor: '#b6a809',
        borderTop: '20px solid #b6a809',
        borderBottom: '20px solid #b6a809',
        borderLeft: '60px solid #b6a809',
        borderRight: '60px solid #b6a809',
        borderRadius: 3,
        display: 'inline-block',
      },
    });

    expect(ReactDomServer.renderToString(<Button />)).toMatchSnapshot();
  });

  afterAll(() => {
    origSpec = JSON.parse(JSON.stringify(Spec));

    patchStrategy(Spec, origSpec, null);
  });
});
