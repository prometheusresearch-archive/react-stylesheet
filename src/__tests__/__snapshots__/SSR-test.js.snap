// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR dynamic-inline strategy hello world button Element  1`] = `"<div style=\\"background:yellow;color:red;box-sizing:border-box;\\" class=\\"rs-0\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"309733784\\"></div>"`;

exports[`SSR dynamic-inline strategy hello world button Style  1`] = `"<div style=\\"font-size:16px;line-height:20;color:#ffffff;text-decoration:none;background-color:#b6a809;border-top:20px solid #b6a809;border-bottom:20px solid #b6a809;border-left:60px solid #b6a809;border-right:60px solid #b6a809;border-radius:3px;display:inline-block;\\" class=\\"div-856143221 rs-0\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"-1622838613\\"></div>"`;
